### PR TITLE
GTJ-98

### PR DESCRIPTION
**Pull request by gulyaasjaanos**
Log component now uses store to display log entries (buildings and troops)
(Log test updated)

### CODE REVIEW

File: backend/migrations/20200729222935-kingdom-troops.js [hunk 1]
```diff
@@ -25,8 +25,8 @@ exports.up = function (db) {
     hp: { type: 'int', notNull: true },
     attack: { type: 'int', notNull: true },
     defence: { type: 'int', notNull: true },
-    started_at: { type: 'timestamp', notNull: true },
-    finished_at: { type: 'timestamp', notNull: true },
+    started_at: { type: "timestamp", notNull: true, defaultValue: new String('CURRENT_TIMESTAMP') },
+    finished_at: { type: "timestamp", notNull: true, defaultValue: new String('CURRENT_TIMESTAMP') }
```
Comment (adamgyulavari) [inline backend/migrations/20200729222935-kingdom-troops.js:29]: @gulyaasjaanos don't modify existing migrations, create a new one and modify fields, though are you sure that this is a good idea?

Comment (gulyaasjaanos) [reply backend/migrations/20200729222935-kingdom-troops.js]: I'm not sure at all, but migration failed without it.

File: backend/src/controllers/SessionController.js [hunk 2]
```diff
@@ -1,40 +1,26 @@
...
-          ? this.errorMessages[error.message].status
-          : 401;
-        const message = this.errorMessages[error.message]
-          ? this.errorMessages[error.message].message
-          : error.message;
-        res.status(status).json({ error: message });
-      });
-  }
-}
+};
```
Comment (adamgyulavari) [inline backend/src/controllers/SessionController.js:26]: @gulyaasjaanos why change the formating?

File: backend/test/api-session.test.js [hunk 3]
```diff
@@ -71,7 +71,7 @@ test('proper data returns token', done => {
     .set('Accept', 'application/json')
     .send({ username: 'superuser', password: 'password' })
     .expect('Content-Type', /json/)
-    .expect(201)
+    .expect(200)
```
Comment (adamgyulavari) [inline backend/test/api-session.test.js:74]: @gulyaasjaanos it should be actually `201`

File: backend/src/controllers/SessionController.js [hunk 4]
```diff
@@ -1,40 +1,26 @@
...
+            [errorCodes.missingUserName]: {status: 401, message: 'Username is reuired.'},
+            [errorCodes.missingPassword]: {status: 401, message: "Password is required."},
+            [errorCodes.invalidUserNameAndPassword]: {status: 401, message: 'Invalid username or password.'},
+        };
+    };
+
+    post(req,res) {
+        const {username,password} = req.body;
+        this.session.login({userName:username,password})
+        .then( response => res.status(200).json(response) )
```
Comment (adamgyulavari) [inline backend/src/controllers/SessionController.js:17]: @gulyaasjaanos it should be `201`

File: frontend/src/components/Log.js [hunk 5]
```diff
@@ -1,23 +1,50 @@
...
 import './Log.css';
 
-const Log = () => {
-  const [data, setData] = useState([]);
+const Log = ({ kingdom, buildings, troops, get }) => {
 
   useEffect(() => {
-    setData(logEntryService.readLogEntries());
-  }, []);
+    get(kingdom);
```
Comment (adamgyulavari) [inline frontend/src/components/Log.js:11]: @gulyaasjaanos let's name this a little better :D
something like `fetchBuildingsAndTroops`

File: frontend/src/components/Log.js [hunk 6]
```diff
@@ -1,23 +1,50 @@
...
 import LogEntry from './LogEntry';
-import { logEntryService } from '../services/logEntryService';
+import { connect } from 'react-redux';
+import { getTroopsAction } from '../actions/TroopsActions';
+import { getBuildingsAction } from '../actions/BuildingsActions';
 import './Log.css';
 
-const Log = () => {
-  const [data, setData] = useState([]);
+const Log = ({ kingdom, buildings, troops, get }) => {
```
Comment (adamgyulavari) [inline frontend/src/components/Log.js:8]: @gulyaasjaanos `kingdom` is just an id, so let's call it `kingdomId`

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**